sys/arduino/base.cpp

```diff
@@ -58,3 +61,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+int analogRead(int adc_pin)
+{
+    int adc_value;
+
+    /* State of ADC channel.
+     * 0: Not initialized
+     * 1: Successfully initialized
+     */
+    static int adc_state[ADC_NUMOF];
+
+    /* Check if the ADC line is valid */
+    if ((adc_pin < 0) || (adc_pin >= ADC_NUMOF)) {
+        return -1;
+    }
+
+    /* Initialization of given ADC channel */
+    if (adc_state[adc_pin] == 0) {
+        if (adc_init(ADC_LINE(adc_pin)) == 0) {
```

Comment:
this is not correct: it is not guaranteed, that the RIOT ADC lines is mapped 'in order' to the Ardunio ADC pins (see `boards/ardunio-due/include/arduino-pinmap.h` as example... The solution here is the same as done for the Arduino digital pins: we need to define a mapping between Ardunio analog pin numbers to actual RIOT ADC lines (as in `boards/arduino-due/include/arduino-board.h`):
```c
static const adc_t arduino_analog_pinmap[] = {
    ARDUINO_A0,
    ....
};
```